tests/rkt_socket_proxyd_test.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	sd_dbus "github.com/coreos/go-systemd/dbus"
+	sd_util "github.com/coreos/go-systemd/util"
+	"github.com/coreos/rkt/tests/testutils"
+	"github.com/vishvananda/netlink"
+)
+
+func TestSocketProxyd(t *testing.T) {
+	if !sd_util.IsRunningSystemd() {
+		t.Skip("Systemd is not running on the host.")
+	}
+
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	iface, _, err := testutils.GetNonLoIfaceWithAddrs(netlink.FAMILY_V4)
+	if err != nil {
+		t.Fatalf("Error while getting non-lo host interface: %v\n", err)
+	}
+	if iface.Name == "" {
+		t.Skipf("Cannot run test without non-lo host interface")
+	}
+
+	nt := networkTemplateT{
+		Name:   "ptp0",
+		Type:   "ptp",
+		IpMasq: true,
+		Master: iface.Name,
+		Ipam: ipamTemplateT{
+			Type:   "host-local",
+			Subnet: "192.168.0.0/24",
+			Routes: []map[string]string{
+				{"dst": "0.0.0.0/0"},
+			},
+		},
+	}
+
+	netDir := prepareTestNet(t, ctx, nt)
+	defer os.RemoveAll(netDir)
+
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
This can be moved just before line 93 since the variable `r` is not used before.